### PR TITLE
Add thread functions

### PR DESCRIPTION
**Pull request by citronneur**
Hi @commial !
I've added some functions around cursor, especially to manipulate thread context.
I've also added default ctor for the Cursor object.

@+

### CODE REVIEW

File: example_api/main.cpp [hunk 1]
```diff
@@ -30,7 +30,7 @@ int main()
 	int result;
 
 	std::cout << "Openning the trace\n";
-	result = ttdengine.Initialize(L"D:\\traces\\demo.run");
```
Comment (commial) [inline example_api/main.cpp:33]: Maybe we can avoid this diff :smiley:

File: TTD/TTD.cpp [hunk 2]
```diff
@@ -30,16 +30,21 @@ namespace TTD {
...
+
 	struct TTD_Replay_ThreadInfo* Cursor::GetThreadInfo() {
 		return this->GetThreadInfo(0);
 	}
 	struct TTD_Replay_ThreadInfo* Cursor::GetThreadInfo(unsigned int ThreadId) {
 		return this->cursor->ICursor->GetThreadInfo(cursor, ThreadId);
 	}
 
-	struct TTD_Replay_RegisterContext* Cursor::GetCrossPlatformContext() {
+	struct TTD_Replay_RegisterContext* Cursor::GetCrossPlatformContext(uint32_t threadId) {
```
Comment (commial) [inline TTD/TTD.cpp:45]: From what I understand from the API, giving a `threadId` of 0 means "current thread".
Do you agree with that?

If so, we can add a wrapper `GetCrossPlatformContext()` wrapping to `GetCrossPlatformContext(0)`.

If not in this PR, I can add it later ;)

**Comment by commial** [comment]
Hey, thanks for the contrib!

File: example_api/main.cpp (hunk 1 above)
Comment (citronneur) [reply example_api/main.cpp]: 💯

File: TTD/TTD.cpp (hunk 2 above)
Comment (citronneur) [reply TTD/TTD.cpp:45]: Exactly, 0 means current thread

**Comment by citronneur** [comment]
I fix my PR. Sorry for the mistakes on example_api/main.cpp, and I've also added the default parameter for Cursor::GetCrossPlatformContext. Thanks!

**Pull request merged**